File: src/main/java/hu/telekom/tkriszti/flightInfo/model/Flight.java
package hu.telekom.tkriszti.flightInfo.model;

import jakarta.persistence.*;

import java.util.List;
import java.util.Set;

@Entity
@Table(name = "flights")
public class Flight {
	@Id
	@Column(name = "id")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int Id;
	@Column(name = "pilot1id")
	private int pilot1Id;
	@Column(name = "pilot2id")
	private int pilot2Id;
	@Column(name = "countryfrom")
	private int countryFrom;
	@Column(name = "countryto")
	private int countryTo;
	private int flightTime; // minutes

	@ManyToMany(mappedBy = "flights")
	private Set<Pilot> pilots;

	public Flight(int Id, int pilot1Id, int pilot2Id, int countryFrom, int countryTo, java.lang.Integer flightTime) {
		this.Id = Id;
		this.pilot1Id = pilot1Id;
		this.pilot2Id = pilot2Id;
		this.countryFrom = countryFrom;
		this.countryTo = countryTo;
		this.flightTime = flightTime;
	}

	public Flight() {
	}

	public int getID() {
		return Id;
	}

	public int getPilot1Id() {
		return pilot1Id;
	}

	public int getPilot2Id() {
		return pilot2Id;
	}

	public int getCountryFrom() {
		return countryFrom;
	}

	public int getCountryTo() {
		return countryTo;
	}

	public java.lang.Integer getFlightTime() {
		return flightTime;
	}

	@Override
	public String toString() {
		return "" + Id;
	}
}

File: src/main/kotlin/hu/telekom/tkriszti/flightInfo/model/Flight.kt
package hu.telekom.tkriszti.flightInfo.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "flights")
data class Flight(
    @Id
    val id: String,
    val pilot1Id: String,
    val pilot2Id: String,
    val countryFrom: String,
    val countryTo: String,
    val flightTime: Int
) {
//import jakarta.persistence.*
//
//@Entity
//@Table(name = "flights")
//class Flight (
//    @Id
//    @Column(name = "id")
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
//    private var Id: Int = 0,
//
//    @Column(name = "pilot1id")
//    private var pilot1Id: Int = 0,
//
//    @Column(name = "pilot2id")
//    private var pilot2Id: Int = 0,
//
//    @Column(name = "countryfrom")
//    private var countryFrom: Int = 0,
//
//    @Column(name = "countryto")
//    private var countryTo: Int = 0,
//
//    @Column(name = "flighttime")
//    private var flightTime: Int = 0, // minutes
//
//    @ManyToMany(mappedBy = "relatedFlights")
//    private val pilots: Set<Pilot>? = null){
//
//    fun getID(): Int {
//        return Id
//    }
//
//    fun getPilot1Id(): Int {
//        return pilot1Id
//    }
//
//    fun getPilot2Id(): Int {
//        return pilot2Id
//    }
//
//    fun getCountryFrom(): Int {
//        return countryFrom
//    }
//
//    fun getCountryTo(): Int {
//        return countryTo
//    }
//
//    fun getFlightTime(): Int {
//        return flightTime
//    }

    override fun toString(): String {
        return "" + id
    }
}